src/patterns.rst

```diff
@@ -570,7 +567,7 @@ A :t:`reference pattern` is a :t:`pattern` that dereferences a :t:`pointer` that
 is being matched.
 
 :dp:`fls_x0bmzl1315gq`
-A :t:`reference pattern` is an :t:`irrefutable pattern`.
+A :t:`reference pattern` is an :t:`irrefutable pattern` when its nested :t:`pattern` is an :t:`irrefutable pattern`.
```

Comment:
I don't understand this sentence

Reply:
This is meant to state that the irrefutable pattern-ness is "inherited". So `&Pattern` is irrefutable when `Pattern` is irrefutable. Any ideas for how to phrase this better? I guess the use of "nested" here is the confusing part?

Reply:
oh, maybe emphasizing *is* would make it more clear...
> A reference pattern is an irrefutable pattern when its nested pattern *is an irrefutable pattern

otherwise it's very clear after you explain it, so feel free to ignore this suggestion

Reply:
alternative to emphasizing "is" is adding "itself"
> A reference pattern is an irrefutable pattern when its nested pattern is itself an irrefutable pattern

Reply:
Added `itself` to it